[JENKINS-63592] Revamp empty states

fqueiruga opened the pull request:
See [JENKINS-63592](https://issues.jenkins-ci.org/browse/JENKINS-63592).

This PR aims to revamp the empty state screens by making them more modern-looking and attractive. They style the empty state items as content blocks in order to provide a stepping stone to make them extensible through an extension point.

#### Screenshots

<details>
<summary>Anonymous user</summary>

Old
![anonymous_old](https://user-images.githubusercontent.com/5738588/92230702-de066200-eeab-11ea-8e9f-f55b7c26503c.png)

New
![anonymous_new](https://user-images.githubusercontent.com/5738588/92230705-df378f00-eeab-11ea-9392-87c64fbd162e.png)
</details>

<details>
<summary>Nothing created - admin user</summary>

Old
![welcome_manage_cloud_old](https://user-images.githubusercontent.com/5738588/92230724-e65e9d00-eeab-11ea-89a1-166adf15a221.png)

New
<img width="1675" alt="create_cloud_new2" src="https://user-images.githubusercontent.com/5738588/92992708-32819100-f4ed-11ea-96d2-812000cd36a6.png">

</details>

<details>
<summary>Nothing created - user with create agent permissions</summary>

Old
![welcome_create_agent_old](https://user-images.githubusercontent.com/5738588/92230787-f8d8d680-eeab-11ea-8b2a-646c46d8774b.png)

New
<img width="1675" alt="create_agent_new2" src="https://user-images.githubusercontent.com/5738588/92992721-3b726280-f4ed-11ea-98dc-b502640b2054.png">

</details>

<details>
<summary>Nothing created - user with only job create permissions</summary>

Old
![welcome_create_job_old](https://user-images.githubusercontent.com/5738588/92230807-ffffe480-eeab-11ea-8531-78c23ecf4d53.png)

New
<img width="1675" alt="create_job_new2" src="https://user-images.githubusercontent.com/5738588/92992735-5f35a880-f4ed-11ea-9b47-b4e6b2ff4ffa.png">
</details>

<details>
<summary>Empty folder</summary>

Old
![folder_old](https://user-images.githubusercontent.com/5738588/92230827-07bf8900-eeac-11ea-97d9-8f04ceab7a08.png)

New
![folder_new](https://user-images.githubusercontent.com/5738588/92370352-772ab800-f0fa-11ea-84e1-bd02d8cd3360.png)
</details>


#### Design brief
Uploaded to the JIRA ticket

#### Breaking changes

This PR makes the previous translations obsolete. All the strings except the welcome message have been removed

### Proposed changelog entries

* Revamp the empty state screens. Their translation strings are now obsolete.

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@daniel-beck 
@timja 
@oleg-nenashev 


<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

fqueiruga committed:
wip - extract conditional vars

fqueiruga committed:
Rework the empty state screends

- Create content block components for the sectioning and hyperlinks
- Reorganize the conditional logic to use the new sectioning

fqueiruga committed:
Set up i18n and clear outdated strings

fqueiruga commented:
I know the git messages are bad but I wasn't able to rebase locally for some reason.

zbynek commented:
Nice improvement! The "Welcome to Jenkins" title seems redundant in the empty folder -- maybe out of scope here but would be nice to remove it in that context.

fqueiruga committed:
Fix broken build for empty state styles

fqueiruga committed:
Undo unintended change to global hyperlinks

- I added a CSS class for generic dark links that should be left for a different PR

fqueiruga committed:
UI touches for the empty state styles

- No longer show welcome message for empty folders
- Update link focus outline color to palette

fqueiruga commented:
I have removed the welcome message from the empty folder screens, you were right.

The PR I think it's now ready and I have taken it out of WIP status

timja commented:
the build is failing for encoding issues in the properties files

fqueiruga committed:
Try to fix encoding issues

fqueiruga commented:
I have fixed those encoding issues

MRamonLeon commented:
Reviewed just the screenshots I see some information has been lost in the revamp, for example _to start building your software project_ Although I believe all of them may be superfluous words, I do think we should have the option to add more information to every _action_ for future options or to improve the knowledge about what the action does. I'm thinking about for example a question mark on every action which has such an extra text. Not showing the question mark if nothing is written. I say a question mark but it's more a UI design decision, what I wouldn't like is a text showing up when you pass with the mouse, to be mobile friendly. WDYT?

fqueiruga commented:
I think you do have a point Ramón. Here are some options that I have sketched:

![Captura de pantalla 2020-09-08 a las 11 33 31](https://user-images.githubusercontent.com/5738588/92459671-43699400-f1c7-11ea-9de9-7ee0f413044d.png)

![Captura de pantalla 2020-09-08 a las 11 33 58](https://user-images.githubusercontent.com/5738588/92459675-44022a80-f1c7-11ea-810d-3cfdae25cfa4.png)

Btw, I appreciate ALL copyediting feedback here.

MRamonLeon commented:
> Btw, I appreciate ALL copyediting feedback here.

I was thinking more about small pieces of text (a sentence or two) for every action attached to them, like the texts that were there previously.

Said that, being able to add a help link (with a different style) to every group or randomly, instead of at every single link/action, is worth as well.

fqueiruga commented:
@Ramón that would be a bit more complex I think.

@timja would you like to see any of the variants on https://github.com/jenkinsci/jenkins/pull/4926#issuecomment-688747407 implemented?

timja commented:
Struggling to tell what's different. It looks good to me currently, but a before / after would help in the screenshots you added

fqueiruga commented:
@timja these are the options compared

<details>
<summary>Before and after</summary>

Old
![welcome_manage_cloud_old](https://user-images.githubusercontent.com/5738588/92230724-e65e9d00-eeab-11ea-89a1-166adf15a221.png)

New
![welcome_manage_cloud_new](https://user-images.githubusercontent.com/5738588/92230732-e8c0f700-eeab-11ea-8987-59b7cf0e027b.png)
</details>

<details>
<summary>Proposed changes</summary>

Section name change:
![Captura de pantalla 2020-09-08 a las 11 33 31](https://user-images.githubusercontent.com/5738588/92459671-43699400-f1c7-11ea-9de9-7ee0f413044d.png)

More info on the link:
![Captura de pantalla 2020-09-08 a las 11 33 58](https://user-images.githubusercontent.com/5738588/92459675-44022a80-f1c7-11ea-810d-3cfdae25cfa4.png)

With link to external docs (would need a page to link to):
![Captura de pantalla 2020-09-11 a las 13 26 55](https://user-images.githubusercontent.com/5738588/92920641-e40fbc00-f432-11ea-9145-766051c1250d.png)
</details>

Which one would you prefer? Also you @MRamonLeon , I have added your suggestion as a third option. I'm partial to the first one myself (changed section title).

oleg-nenashev commented:
IMHO "Create a job" should go ahead of "Create a distributed build". Even if it is not advised, I believe the most of instances begin from running jobs on the controller

timja commented:
Any thoughts on whether the states are too far to the right?

On my monitor at full screen:
![image](https://user-images.githubusercontent.com/21194782/92975086-72119400-f47f-11ea-8d7c-4c02067dabbe.png)

Half size:

![image](https://user-images.githubusercontent.com/21194782/92975116-82297380-f47f-11ea-9497-838577402d6c.png)

My monitor is reasonably large (27") but it does seem a bit funny.

timja commented on core/src/main/resources/hudson/model/AllView/noJob.groovy:
```diff
@@ -4,40 +4,142 @@ import hudson.model.Computer
+
+                            if (hasAdministerJenkinsPermission) {
+                                li(class: "content-block") {
+                                    a(href: "computer/new", class: "content-block__link") {
```
Hmm this doesn't seem right,

1. it's just linking to new computer, not the cloud configuration screen.
2. it's showing up when I don't have a cloud plugin installed

Although 2 probably isn't too bad as there's docs on the configure Clouds page saying what to do:

![image](https://user-images.githubusercontent.com/21194782/92975381-07ad2380-f480-11ea-8410-b98c1a20ae77.png)

```suggestion
                                    a(href: "configureClouds", class: "content-block__link") {
```

fqueiruga replied:
Maybe I messed something up

fqueiruga committed:
Update core/src/main/resources/hudson/model/AllView/noJob.groovy

Co-authored-by: Tim Jacomb <timjacomb1+github@gmail.com>

fqueiruga commented:
The content is centered within the main panel. Not sure what the best approach would be

timja commented:
I think it’s the side panel that makes it look weird on a large screen

The balance seems a bit off.

Not blocking though

fqueiruga commented:
Not much we can do about that one.

fqueiruga committed:
Move create job to top and change section title

fqueiruga commented:
I have moved "Create a job" to the top as per Oleg's suggestion and I have also changed it's section title with copy from the former implementation.

daniel-beck commented:
The introductory text for

> Nothing created - user with only job create permissions

mentions distributed builds, but the action is (correctly) unavailable. This seems slightly confusing.

oleg-nenashev reviewed (approved):
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process
